classes/01class/exercises/c01-aws01/zarajoy/README.md

```diff
@@ -0,0 +1,393 @@
+                "Code": "pending",
+                "Message": "pending"
+            },
+            "VirtualizationType": "hvm",
+            "CpuOptions": {
+                "CoreCount": 1,
+                "ThreadsPerCore": 1
+            },
+            "CapacityReservationSpecification": {
+                "CapacityReservationPreference": "open"
+            },
+            "MetadataOptions": {
+                "State": "pending",
+                "HttpTokens": "optional",
+                "HttpPutResponseHopLimit": 1,
+                "HttpEndpoint": "enabled"
+            }
+        }
+    ],
+    "OwnerId": "512742231244",
+    "ReservationId": "r-07cc8f6e71d7afe6e"
+}
+```
+
+- Commands to connect to the first EC2 instance:
+```
+z@bacon:~$ ssh -i awskp01.pem ec2-user@13.239.28.37
+Warning: Identity file awskp01.pem not accessible: No such file or directory.
+The authenticity of host '13.239.28.37 (13.239.28.37)' can't be established.
+ECDSA key fingerprint is SHA256:ewimNRcdoLHVsDfJ4iebEcWD920WprRuJxRiQOZkphw.
+Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
+Warning: Permanently added '13.239.28.37' (ECDSA) to the list of known hosts.
+ec2-user@13.239.28.37: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).
+z@bacon:~$ aws ec2 describe-instances --query "Reservations[].Instances[].[InstanceId, PublicIpAddress]"
+[
+    [
+        "i-07b1ba8023f12e889",
+        null
+    ],
+    [
+        "i-00cb16d60056e7353",
+        "13.239.28.37"
+    ]
+]
+z@bacon:~$ ssh -i awskp01.pem ec2-user@13.239.28.37
+Warning: Identity file awskp01.pem not accessible: No such file or directory.
+z@bacon:~$ sudo chmod 400 aws01kp.pem
+[sudo] password for z: 
+z@bacon:~$ ssh -i aws01kp.pem ec2-user@13.239.28.37
+
+       __|  __|_  )
+       _|  (     /   Amazon Linux 2 AMI
+      ___|\___|___|
+
+https://aws.amazon.com/amazon-linux-2/
+4 package(s) needed for security, out of 8 available
+Run "sudo yum update" to apply all updates.
+[ec2-user@ip-172-31-36-157 ~]$ 
+
+```
+
+- Create another EC2, also using the CLI instance in the same VPC but with private address only and connect to it.
+- Commands to create the second EC2 instance and any additional resource required:
+```
+z@bacon:~$ aws ec2 create-security-group --group-name SecureGroupv2 --description  "Secure group for DOA"
+{
+    "GroupId": "sg-041c7d3e1c204e2df"
+}
+z@bacon:~$ aws ec2 authorize-security-group-ingress --group-name SecureGroupv2 --protocol tcp --port 22 --cidr 172.31.36.157/32
+z@bacon:~$ aws ec2 describe-security-groups --group-name SecureGroupv2
+{
+    "SecurityGroups": [
+        {
+            "Description": "Secure group for DOA",
+            "GroupName": "SecureGroupv2",
+            "IpPermissions": [
+                {
+                    "FromPort": 22,
+                    "IpProtocol": "tcp",
+                    "IpRanges": [
+                        {
+                            "CidrIp": "172.31.36.157/32"
```

Comment:
Nice use of least privilege. For security groups, you can also configure the source as another Security group, so you could also configure this security group to allow only connects on port 22 from the Security group of the public instance and you would have a similar effect in this case.